### DISCUSSION TITLE
rerun.sh file is not generated

### DISCUSSION THREAD

**Discussion by firmansyahrosid** [webdriverio-community/wdio-rerun-service#70] [General]
Hi,

I used the latest version 2.0.7, but the rerun.sh file is not generated.
This is my configuration:

```
import RerunService from "wdio-rerun-service";

services: [
    [
      RerunService,
      {
        rerunDataDir: join(process.cwd(), "./reports/rerun"),
        rerunScriptPath: join(process.cwd(), "./reports/rerun.sh"),
        commandPrefix: "VARIABLE=true"
      },
    ],
  ]
```

**Comment by Emmet-McGuire** [comment]
@firmansyahrosid  did you figure this out?

I am trying to upgrade from WDIO v7 to v8, and wdio-rerun-service (2.0.7) is not generating the rerun.sh file.

This has worked great for a long time for me, on the older version (to go w/ WDIO v7).

**Comment by mikesalvia** [comment]
@Emmet-McGuire can you post your versions? I am using V8 and I see a `rerun.sh` file being generated.

**Reply by Emmet-McGuire** [reply]
Here's actually a repo demonstrating the issue:  https://github.com/Emmet-McGuire/wdio-rerun-service-v8-example/tree/main

```
"dependencies": {
        "@wdio/cli": "^8.16.12",
        "@wdio/cucumber-framework": "^8.16.12",
        "@wdio/local-runner": "^8.16.12",
        "@wdio/spec-reporter": "^8.16.12",
        "chromedriver": "^117.0.3",
        "wdio-chromedriver-service": "^8.1.1",
        "wdio-rerun-service": "^2.0.7"
    },
```

**Reply by OluJS** [reply]
Are there any updates on a fix for this?

**Reply by Emmet-McGuire** [reply]
@OluJS, I haven't seen any.  I had to switch to the [default webdriver.io retry capability](https://webdriver.io/docs/retry/#rerunning-in-cucumber), which has drawbacks for cucumber.

**Reply by firmansyahrosid** [reply]
Hi @mikesalvia any updates for this ?

**Reply by gustavo-pequeno** [reply]
any updates about this issue?

**Reply by seanpoulter** [reply]
I'd encourage you to work together to troubleshoot the issue and open a PR. The docs for [custom services](https://webdriver.io/docs/customservices/) are a good place to start to understand how the service is structured. From there I'd recommend using the `--inspect` arg documented on [this page](https://webdriver.io/docs/debugging) to step through the code with your debugger.

You'll also may want to collaborate with @Emmet-McGuire to update the example for v8 since the `wdio-chromedriver-service` has been replaced by `webdriverio` and will fail to download the latest version of ChromeDriver. (The URL changed so there's no updates 💩 .)

**Comment by pheeel** [comment]
I noticed that if you have a `before` block in the spec file and your test falls just in that block, that test file will not be written to the rerun.sh file.
